buildchain/buildchain/targets/local_image.py

```diff
@@ -162,9 +162,20 @@ def check_dockerfile_dependencies(self) -> Union[TaskError, bool]:
             return TaskError(msg=error_message)
         return None
 
+    def clean_registry(self) -> Optional[TaskError]:
+        """Docker API call for removing the image from local registry."""
+        docker_remove = docker_command.DockerRMI(self.name, self.version)
+        return docker_remove()
+
     @property
     def task(self) -> types.TaskDict:
         task = self.basic_task
+
+        if self.save_on_disk:
```

Comment:
```
clean =  [self.clean_registry]
if self.save_on_disk:
    clean.append(super().clean)
```

Reply:
I'd rather have an explicit setup of the `self.save_on_disk` divergences at the cost of very minor code duplication.

Reply:
But it's less clear.
In both case we do need to clear the image on the registry and iif we save it on disk then we also need to clean it on disk.